### ISSUE TITLE
Latest update FOREIGN KEY constraint failed_

### ISSUE THREAD

**Issue by promexio** [frappe/books#628]
... do you have also this issue?

**Error**: _DatabaseError: Error
failed migration
FOREIGN KEY constraint failed_

**Stack**:
```
Error: failed migration
FOREIGN KEY constraint failed
    at client.transaction.connection (/Applications/Frappe Books.app/Contents/Resources/app.asar/node_modules/knex/lib/execution/runner.js:254:23)
```

**Version**: `0.14.1`
**Platform**: `darwin`
**Path**: `/index.html`
**Language**: `German`

**Comment by 18alantom** [comment]
@promexio I did not face this on my databases.

Any steps to reproduce or a test database which will cause this issue?

**Comment by promexio** [comment]
Sorry, got it working by importing backup table and then migrate data. 
The problematic table was Party PK =party.name
It's not 100% sure that this was due to FBOOKS ---could also be that I did some deep changes previously... 

Now, I have only an issue if I open the reports section: Do you have any idea?

**Error**: _TypeError: Cannot set properties of undefined (setting 'prune')_

**Stack**:
```
TypeError: Cannot set properties of undefined (setting 'prune')
    at j (app://./js/index.d5da6bbd.js:8:387668)
    at v (app://./js/index.d5da6bbd.js:8:387610)
    at Proxy._getAccountTree (app://./js/index.d5da6bbd.js:8:383660)
    at async Proxy.setReportData (app://./js/index.d5da6bbd.js:8:389529)
    at async Proxy.initialize (app://./js/index.d5da6bbd.js:8:446342)
    at async Proxy.setReportData (app://./js/index.d5da6bbd.js:8:272191)
    at async Proxy.activated (app://./js/index.d5da6bbd.js:8:271321)
```

**Version**: `0.14.1`
**Platform**: `darwin`
**Path**: `/report/ProfitAndLoss`
**Language**: `German`
**Country**: `de`

**Comment by promexio** [comment]
It's really bizarre: I can't see the Chart of Accounts, but it's in the table Account of the database. 
And even if I look-up in a general journal, I can retrieve the data. 
![image](https://github.com/frappe/books/assets/112803939/5f101149-5cbb-49d9-b6f7-5aedd576f0cb)

![image](https://github.com/frappe/books/assets/112803939/7087a402-1d5d-49aa-ac3a-fd70e7de7f72)

**Comment by promexio** [comment]
Solved! I got it working. 
The problem was that I manipulated the CoA in mass via DBeaver and [null] was a blank in the db. Removed and replaced with NULL ---working!
As the reports are views applying filters based on the CoA, it's normal that they weren't working anymore.
![image](https://github.com/frappe/books/assets/112803939/40fd5868-c8b8-4b49-8f8c-744ab083ecfb)

Finding the error (behind the wheel) helped me a lot to understand how FBOOKS works.
I would also be curious if it's possible to develop new reports easily ...? How? Is there any documentation?
